walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,77 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.sps.TimeRange;
+import java.util.Comparator;
+import com.google.sps.Event;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        System.out.println("THen end of the day is: " + TimeRange.END_OF_DAY);
+        long duration = request.getDuration();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+        Collection<String> mandatoryAttendees = request.getAttendees();
+        Collection<String> allAttendees = new ArrayList<String>();
+        Collection<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+        for (String attendee : mandatoryAttendees){
+            allAttendees.add(attendee);
+        }
+        for (String attendee : optionalAttendees){
+            allAttendees.add(attendee);
+        }
+        if (mandatoryAttendees.isEmpty()){
+            meetingTimes = returnMeetingTimes(events, optionalAttendees, duration);
+        }
+        else{
+            meetingTimes = returnMeetingTimes(events, allAttendees, duration).isEmpty() ? 
+            returnMeetingTimes(events, mandatoryAttendees, duration) : returnMeetingTimes(events, allAttendees, duration);
+        }
+        return (meetingTimes);
+    }
+
+    public boolean attendeesAtEvent(Event event, Collection<String> attendees){
```

Comment:
I see you are only testing the query method, should this and below methods be private?